unimport/refactor.py

```diff
@@ -1,137 +1,111 @@
+        location = self.get_location(original_node)
+        if isinstance(updated_node.names, cst.ImportStar):
+            if isinstance(updated_node.module, cst.Attribute):
+                import_name = self.get_import_name_from_attr(
+                    attr_node=updated_node.module
+                )
+            else:
+                import_name = updated_node.module.value
+            imp = self.get_imp(import_name=import_name, location=location)
+            if imp["modules"]:
+                modules = ",".join(imp["modules"])
+                names_to_suggestion = []
+                for module in modules.split(","):
+                    names_to_suggestion.append(
+                        cst.ImportAlias(cst.Name(module))
+                    )
+                return updated_node.with_changes(names=names_to_suggestion)
+            else:
+                if imp["module"]:
+                    return cst.RemoveFromParent()
+        else:
+            names_to_keep = []
+            for import_alias in updated_node.names:
+                if isinstance(import_alias.name, cst.Attribute):
+                    import_name = self.get_import_name_from_attr(
+                        attr_node=import_alias.name
+                    )
+                else:
+                    import_name = (
+                        import_alias.asname or import_alias
+                    ).name.value
+                if self.is_import_used(import_name, location):
+                    names_to_keep.append(
+                        import_alias.with_changes(
+                            comma=cst.MaybeSentinel.DEFAULT
+                        )
+                    )
+            if len(names_to_keep) == 0:
+                return cst.RemoveFromParent()
+            else:
+                return updated_node.with_changes(names=names_to_keep)
+        return updated_node
+
+    def leave_Import(
+        self, original_node: cst.Import, updated_node: cst.Import
+    ) -> cst.Import:
+        return self.leave_import_alike(original_node, updated_node)
 
-class RefactorTool(RefactoringTool):
-    def __init__(self):
-        self._fixer = RefactorImports()
-        self._fixers = [self._fixer]
-        super().__init__(None, options={"print_function": True})
+    def leave_ImportFrom(
+        self, original_node: cst.ImportFrom, updated_node: cst.ImportFrom
+    ) -> cst.ImportFrom:
+        return self.leave_import_alike(original_node, updated_node)
 
-    def get_fixers(self):
-        return self._fixers, []
 
-    def refactor_string(self, data, unused_imports, name="unimport"):
-        with self._fixer.clean(unused_imports):
-            source = super().refactor_string(data, name)
-        return str(source)
+def refactor_string(scanner):
+    wrapper = MetadataWrapper(cst.parse_module(scanner.source))
+    fixed_module = wrapper.visit(RemoveUnusedImportTransformer(scanner))
+    return fixed_module.code
```

Comment:
As I said, this shouldn't take up the scanner, it should take the unused imports and the source

Reply:
fix